### PR TITLE
Problems api, database setup, fixed husky problems

### PR DESCRIPTION
**Pull request by Tamas-Leung**

### CODE REVIEW

**Comment by AntonKanug** [comment]
closes #19 
closes #20

**Review by AntonKanug — APPROVED** [review]

File: src/backend/src/database/mongoose.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    Backtracking: "Back Tracking",
+    Graphs: "Graphs",
+    DynamicProgramming: "Dynamic Programming",
+    Greedy: "Greedy",
+    Intervals: "Intervals",
+    MathAndGeometry: "Math And Geometry",
+    BitManipulation: "Bit Manupulation"
+}
+
+const ProblemSchema = new Schema({
```
Comment (YoussefRizkalla) [inline src/backend/src/database/mongoose.js:35]: Have the schemas in a different file and import them

File: src/backend/src/database/mongoose.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+connect(process.env.MONGODB_URL, {
+})
+
+const ProblemDifficulty = {
+    Easy: "Easy",
+    Medium: "Medium",
+    Hard: "Hard"
+}
+
+const ProblemType = {
```
Comment (YoussefRizkalla) [inline src/backend/src/database/mongoose.js:15]: Include in same file as schema

File: src/backend/src/database/mongoose.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
+import { Schema, connect, model } from 'mongoose';
+import dotenv from 'dotenv'
+dotenv.config()
+import m2s from 'mongoose-to-swagger';
+
+connect(process.env.MONGODB_URL, {
+})
+
+const ProblemDifficulty = {
```
Comment (YoussefRizkalla) [inline src/backend/src/database/mongoose.js:9]: Include in same file as schema

**Review by YoussefRizkalla — CHANGES_REQUESTED** [review]

**Review by YoussefRizkalla — APPROVED** [review]

**Review by dipesube — APPROVED** [review]

**Comment by AntonKanug** [comment]
closes #21

**Pull request merged**